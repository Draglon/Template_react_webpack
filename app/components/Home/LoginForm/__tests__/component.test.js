import React from 'react';
import { Formik } from 'formik';

import LoginFormComponent from '../component';

describe('<LoginFormComponent />', () => {
  const props = {
    onSubmit: jest.fn(),
    loginError: '',
  };
  const component = shallow(<LoginFormComponent {...props} />);

  describe('request is loading', () => {
    it('should match snapshot', () => {
      const propsFormik = {
        isSubmitting: true,
      };
      const rendered = component.find(Formik).renderProp('render')(propsFormik);
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('request was complited', () => {
    it('should match snapshot', () => {
      const propsFormik = {
        isSubmitting: false,
      };
      const rendered = component.find(Formik).renderProp('render')(propsFormik);
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('fields have error', () => {
    beforeEach(() => {
      const newProps = {
        ...props,
        loginError: 'field with error',
      };
      component.setProps(newProps);
    });

    it('should match snapshot', () => {
      const propsFormik = {
        isSubmitting: false,
      };
      const rendered = component.find(Formik).renderProp('render')(propsFormik);
      expect(rendered).toMatchSnapshot();
    });
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
