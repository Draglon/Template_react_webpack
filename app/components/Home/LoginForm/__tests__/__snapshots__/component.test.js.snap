// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginFormComponent /> fields have error should match snapshot 1`] = `
<Form
  className="form login__form"
>
  <FormikConnect(FieldInner)
    component={[Function]}
    fieldPrefix={
      <Icon
        style={
          Object {
            "color": "rgba(0,0,0,.25)",
          }
        }
        type="user"
      />
    }
    name="username"
    placeholder="Username"
    type="text"
  />
  <FormikConnect(FieldInner)
    component={[Function]}
    fieldPrefix={
      <Icon
        style={
          Object {
            "color": "rgba(0,0,0,.25)",
          }
        }
        type="lock"
      />
    }
    name="password"
    placeholder="Password"
    type="password"
  />
  <FormButton
    htmlType="submit"
    icon=""
    size="default"
    type="primary"
    value="Log in"
  />
  <div
    className="login__error"
  >
    Pleace, check your login or password and try again
  </div>
</Form>
`;

exports[`<LoginFormComponent /> request is loading should match snapshot 1`] = `
<Form
  className="form login__form"
>
  <FormikConnect(FieldInner)
    component={[Function]}
    fieldPrefix={
      <Icon
        style={
          Object {
            "color": "rgba(0,0,0,.25)",
          }
        }
        type="user"
      />
    }
    name="username"
    placeholder="Username"
    type="text"
  />
  <FormikConnect(FieldInner)
    component={[Function]}
    fieldPrefix={
      <Icon
        style={
          Object {
            "color": "rgba(0,0,0,.25)",
          }
        }
        type="lock"
      />
    }
    name="password"
    placeholder="Password"
    type="password"
  />
  <FormButton
    htmlType="submit"
    icon="loading"
    size="default"
    type="primary"
    value="Log in"
  />
</Form>
`;

exports[`<LoginFormComponent /> request was complited should match snapshot 1`] = `
<Form
  className="form login__form"
>
  <FormikConnect(FieldInner)
    component={[Function]}
    fieldPrefix={
      <Icon
        style={
          Object {
            "color": "rgba(0,0,0,.25)",
          }
        }
        type="user"
      />
    }
    name="username"
    placeholder="Username"
    type="text"
  />
  <FormikConnect(FieldInner)
    component={[Function]}
    fieldPrefix={
      <Icon
        style={
          Object {
            "color": "rgba(0,0,0,.25)",
          }
        }
        type="lock"
      />
    }
    name="password"
    placeholder="Password"
    type="password"
  />
  <FormButton
    htmlType="submit"
    icon=""
    size="default"
    type="primary"
    value="Log in"
  />
</Form>
`;

exports[`<LoginFormComponent /> should match snapshot 1`] = `
<Layout
  className="login"
>
  <Title>
    The Movie DB
  </Title>
  <Formik
    enableReinitialize={false}
    initialValues={
      Object {
        "password": "",
        "username": "",
      }
    }
    isInitialValid={false}
    onSubmit={[MockFunction]}
    render={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": undefined,
        "_nodes": Array [
          "password",
          "username",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "password": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "username": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
              "undefined": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  />
</Layout>
`;
